src/main/java/graphql/introspection/Introspection.java

```diff
@@ -174,18 +198,41 @@ public enum TypeKind {
                         : null;
             }
             return null;
-        });
-        register(__InputValue, "isDeprecated", environment -> {
+        };
+        IntrospectionDataFetcher<?> isDeprecatedDataFetcher = environment -> {
             Object type = environment.getSource();
             if (type instanceof GraphQLArgument) {
                 return ((GraphQLArgument) type).isDeprecated();
             } else if (type instanceof GraphQLInputObjectField) {
                 return ((GraphQLInputObjectField) type).isDeprecated();
             }
             return null;
-        });
+        };
+        IntrospectionDataFetcher<?> typeDataFetcher = environment -> {
+            Object type = environment.getSource();
+            if (type instanceof GraphQLArgument) {
+                return ((GraphQLArgument) type).getType();
+            } else if (type instanceof GraphQLInputObjectField) {
+                return ((GraphQLInputObjectField) type).getType();
+            }
+            return null;
+        };
+        IntrospectionDataFetcher<?> deprecationReasonDataFetcher = environment -> {
+            Object type = environment.getSource();
+            if (type instanceof GraphQLArgument) {
+                return ((GraphQLArgument) type).getDeprecationReason();
+            } else if (type instanceof GraphQLInputObjectField) {
+                return ((GraphQLInputObjectField) type).getDeprecationReason();
+            }
+            return null;
+        };
+
         register(__InputValue, "name", nameDataFetcher);
         register(__InputValue, "description", descriptionDataFetcher);
+        register(__InputValue, "type", typeDataFetcher);
+        register(__InputValue, "defaultValue", defaultValueDataFetcher);
+        register(__InputValue, "isDeprecated", isDeprecatedDataFetcher);
+        register(__InputValue, "deprecationReason", deprecationReasonDataFetcher);
     }
```

Comment:
I reorganised the registration to that the `register` call is done in simple lines and the fetchers are defined above.

I also ordered them in the order of field definition from above.

So you can see that we defined name field first, so its DF registration is first and so on.